src/xtgeo/grid3d/_egrid.py

```diff
@@ -679,31 +692,31 @@ def mapaxes(self) -> MapAxes | None:
         return self.egrid_head.mapaxes
 
     @mapaxes.setter
-    def mapaxes(self, value):
+    def mapaxes(self, value: MapAxes) -> None:
         self.egrid_head.mapaxes = value
 
     @property
-    def dimensions(self) -> tuple[int, int, int]:
+    def dimensions(self) -> tuple[int, int, int] | None:
         return self.global_grid.dimensions
 
     @property
     def map_axis_units(self) -> Units:
         return self.egrid_head.mapunits
 
     @map_axis_units.setter
-    def map_axis_units(self, value):
+    def map_axis_units(self, value: Units) -> None:
         self.egrid_head.mapunits = value
 
     @property
-    def grid_units(self) -> Units:
+    def grid_units(self) -> Units | None:
```

Comment:
I think this defaults to `Units.METRES` now, so no need for `Optional` return type.